core/src/main/java/com/sequenceiq/cloudbreak/service/stack/StackImageService.java

```diff
@@ -114,4 +120,29 @@ private void replaceStackImageComponent(Stack stack, StatedImage targetImage) {
         LOGGER.debug("The following target image found for stack {}, {}", stackId, targetImageComponent);
         return Optional.ofNullable(targetImageComponent);
     }
+
+    public Optional<StatedImage> getStatedImageInternal(Stack stack, com.sequenceiq.cloudbreak.cloud.model.Image image, ImageCatalog imageCatalog) {
+        return ThreadBasedUserCrnProvider.doAs(
+                internalCrnModifier.getInternalCrnWithAccountId(Crn.fromString(stack.getResourceCrn()).getAccountId()),
+                () -> {
+                    try {
+                        return Optional.ofNullable(imageCatalogService.getImageByCatalogName(
+                                stack.getWorkspace().getId(), image.getImageId(), imageCatalog.getName()));
+                    } catch (Exception e) {
+                        return Optional.empty();
+                    }
+                });
+    }
+
+    public Optional<StatedImage> getStatedImageForStackInternal(Stack stack) throws CloudbreakImageNotFoundException {
+        com.sequenceiq.cloudbreak.cloud.model.Image image = getCurrentImage(stack);
+        ImageCatalog imageCatalog = getImageCatalogFromStackAndImageInternal(stack, image);
+        return getStatedImageInternal(stack, image, imageCatalog);
+    }
+
+    public ImageCatalog getImageCatalogFromStackAndImageInternal(Stack stack, com.sequenceiq.cloudbreak.cloud.model.Image image) {
+        return ThreadBasedUserCrnProvider.doAs(
+                internalCrnModifier.getInternalCrnWithAccountId(Crn.fromString(stack.getResourceCrn()).getAccountId()),
```

Comment:
I would move this to a separate line. a bit more readable. or maybe create a method for this, as it's the same as in `getStatedImageInternal`